solr/jetty/src/java/org/apache/solr/jetty/QueuedThreadPool.java

```diff
@@ -0,0 +1,1178 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.apache.solr.jetty;
```

Comment:
Curious, have made any change to this file for our purpose?

Reply:
Yes. The commits are logically split to clarify what's pulled in from upstream and different changes -- functional ([48c971f](48c971fc77bbb7cb95e387ff35618924817a051e)) and rote ([a5a039b](https://github.com/cowpaths/fullstory-solr/pull/79/commits/a5a039bd56d3933e0e0668911cad35d8af46de1b)). This is complicated a bit while the upstream PR is in-flight, because the substance of the changes are in the first of these two commits, which (maybe misleadingly) refers to "changes from upstream Jetty PR". To clarify, these changes are not yet merged upstream, but we're trying to track here the changes as proposed in the upstream PR.